source/common/upstream/health_checker_impl.cc

```diff
@@ -102,6 +105,7 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onInterval() {
       {Http::Headers::get().Path, parent_.path_},
       {Http::Headers::get().UserAgent, Http::Headers::get().UserAgentValues.EnvoyHealthChecker}};
 
+  parent_.request_headers_parser_->evaluateHeaders(request_headers, RequestInfo::RequestInfoImpl());
```

Comment:
nit: Can you allocate a `static const RequestInfoImpl` and use that? Then we can avoid allocating a temporary each time.

Reply:
Sure, added in e28f72d. Thanks.